int16[] qi_set
int16[] q_set
---
bool success